docs/guides/serverless-shared-migration-guide.md

```diff
@@ -0,0 +1,266 @@
+---
+page_title: "Migration Guide: Transition out of Serverless Instances and Shared-tier clusters"
+---
+
+# Migration Guide: Transition out of Serverless Instances and Shared-tier clusters
+
+The goal of this guide is to help users transition from Serverless Instances and Shared-tier clusters (M2/M5) to Free, Flex or Dedicated Clusters. 
+
+Starting in March 2025, all Shared-tier clusters (in both `mongodbatlas_cluster` and `mongodbatlas_advanced_cluster`) will automatically convert to Flex clusters. Similarly, all Serverless instances (`mongodb_serverless_instance`) will be converted into Free/Flex/Dedicated clusters, [depending on your existing configuration](https://www.mongodb.com/docs/atlas/flex-migration/). 
+These conversions will result in configuration drift in Terraform. 
+
+
+You can migrate from Serverless instances and Shared-tier clusters manually before March 2025. 
+
+**--> NOTE:** We recommend waiting until March 2025 or later for Serverless instances and Shared-tier clusters to autoconvert.
+
+If a Serverless instance has $0 MRR, it automatically converts into a Free cluster. Else, if it does not fit the constraints of a Flex cluster, it will convert into a Dedicated cluster, resulting in downtime and workload disruption. Otherwise, it will convert to a Flex cluster.
+
+For Shared-tier clusters, we are working on enhancing the User Experience such that Terraform Atlas Providers users can make even fewer required changes to their scripts from what is shown below. More updates to come over the next few months. For more details reach out to: zuhair.ahmed@mongodb.com
+
+### Jump to:
+- [Shared-tier to Flex](#from-shared-tier-clusters-to-flex)
+- [Serverless to Free](#from-serverless-to-free)
+- [Serverless to Flex](#from-serverless-to-flex)
+- [Serverless to Dedicated](#from-serverless-to-dedicated)
+
+## From Shared-tier clusters to Flex 
+### Pre-Autoconversion Migration Procedure
+
+**NOTE:** We recommend waiting until March 2025 or later for Shared-tier clusters to autoconvert.
```

Comment:
doesn't this cause downtime? should that be clear?

Reply:
I am not aware of this auto conversion to cause downtime, at least I haven't seen it in any docs. The one auto conversion that will cause downtime that I know of is Serverless to Dedicated

Reply:
I am referrring to when they do it manually, if they are migrating manually shouldn't we warn them it causes downtime? It requires creating a new cluster and migrate data..

Reply:
RIght, I made it explicit in all the pre-autoconvestion sections in this commit https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2746/commits/c5f4719ed46345e27e0ca0a2880c6bd66cdb21b6